app/src/main/java/org/p2p/wallet/auth/gateway/repository/mapper/GatewayServiceUpdateMetadataMapper.kt

```diff
@@ -0,0 +1,73 @@
+package org.p2p.wallet.auth.gateway.repository.mapper
+
+import com.google.gson.Gson
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+import org.p2p.core.rpc.JsonRpc
+import org.p2p.solanaj.utils.crypto.toBase64Instance
+import org.p2p.wallet.auth.gateway.api.request.GatewayOnboardingMetadataCiphered
+import org.p2p.wallet.auth.gateway.api.request.UpdateMetadataRpcRequest
+import org.p2p.wallet.auth.gateway.api.response.UpdateMetadataResponse
+import org.p2p.wallet.auth.gateway.repository.model.GatewayOnboardingMetadata
+import org.p2p.wallet.utils.Base58String
+import org.p2p.wallet.utils.toByteArray
+import org.p2p.wallet.utils.toJsonObject
+
+class GatewayServiceUpdateMetadataMapper(
+    private val onboardingMetadataCipher: GatewayServiceOnboardingMetadataCipher,
+    private val signatureFieldGenerator: PushServiceSignatureFieldGenerator,
+    private val gson: Gson
+) {
+
+    private class UpdateMetadataStruct(
+        val ethereumAddress: String,
+        val solanaPublicKey: String,
+        val epochUnixTime: Long,
+        val encryptedMetadata: String,
+    ) : BorshSerializable {
+        override fun serializeSelf(): ByteArray {
+            return getBorshBuffer()
+                .write(ethereumAddress, solanaPublicKey, epochUnixTime, encryptedMetadata)
```

Comment:
I can't see the 4 field in the specification. Request to update it

Reply:
https://p2pvalidator.atlassian.net/wiki/spaces/Wallet/pages/2479030273/API+Gateway+Get+metadata#Input-parameters.1

struct GetMetadataMessage<'a> {
    ethereum_id: &'a str,
    solana_pubkey: &'a str,
    timestamp_device: i64,
    metadata &'a [u8]
}